engine/battle/stats_functions.asm

```diff
@@ -460,6 +459,16 @@ SetExpAllFlags:
 	pop bc
 	jr .gainExpFlagsLoop
 .return
+	ld a, [wPartyCount]
+	ld c, a
+	ld a, 6
+	sub c
+.loopBitRotation
+	jp z, .finishVariableBitRotation
+	rr b
+	dec a
+	jr .loopBitRotation
+.finishVariableBitRotation
```

Comment:
This wont work. Imagine a party of 6 and this is your first time in the loop. A becomes 6, C becomes 6, and your result is zero so you do two rr commands. The carry value starts in the carry bit. The first rr moves it to bit 7. The second rr moves it to bit 6. wPartyCount is also not decrementing so on the next wMon loop you are always doing 6 - 6. The second mon has its flag value now in bit 6 and the first mon has its flag value now in bit 4.

Reply:
Did you account for the rr b instructions before the loop?